### PR TITLE
ImplicitCreation atribute tests, value class flag verification

### PR DESCRIPTION
**Pull request by theresa-m**
- class verification check to make sure ImplicitCreation attribute is in a non-abstract value class
  - tests for ImplicitCreation class verification

Related to: https://github.com/eclipse-openj9/openj9/issues/17234

### CODE REVIEW

**Comment by hangshao0** [comment]
Could you add a bit more description of the change in the commit message and https://github.com/eclipse-openj9/openj9/pull/17976#issue-1855470773 ?

File: runtime/nls/cfre/cfrerr.nls [hunk 1]
```diff
@@ -1750,3 +1750,10 @@ J9NLS_CFR_ERR_NULLRESTRICTED_ATTRIBUTES_LENGTH_IS_ZERO.explanation=Please consul
 J9NLS_CFR_ERR_NULLRESTRICTED_ATTRIBUTES_LENGTH_IS_ZERO.system_action=The JVM will throw a verification or classloading related exception such as java.lang.ClassFormatError.
 J9NLS_CFR_ERR_NULLRESTRICTED_ATTRIBUTES_LENGTH_IS_ZERO.user_response=Contact the provider of the classfile for a corrected version.
 # END NON-TRANSLATABLE
+
+J9NLS_CFR_ERR_ILLEGAL_CLASS_MODIFIERS=Illegal class modifiers
```
[the remark this answers is not in the document]
Comment (theresa-m) [reply runtime/nls/cfre/cfrerr.nls:1754]: I did this one to match the RI, I can make it more descriptive.

File: runtime/bcutil/cfreader.c [hunk 2]
```diff
@@ -2632,6 +2632,11 @@ checkAttributes(J9PortLibrary* portLib, J9CfrClassFile* classfile, J9CfrAttribut
 #endif /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
 #if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
 		case CFR_ATTRIBUTE_ImplicitCreation:
+			if (J9_ARE_NO_BITS_SET(classfile->accessFlags, CFR_ACC_VALUE_TYPE)) {
```
[the remark this answers is not in the document]
Comment (theresa-m) [reply runtime/bcutil/cfreader.c:2635]: I interpreted this to be anything other then a `non-abstract value class` (I did miss the non-abstract part). Are there any cases this wouldn't cover?

File: runtime/nls/cfre/cfrerr.nls (hunk 1 above)
[the remark this answers is not in the document]
Comment (hangshao0) [reply runtime/nls/cfre/cfrerr.nls:1754]: It is more helpful have a more descriptive message.

File: runtime/bcutil/cfreader.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (hangshao0) [reply runtime/bcutil/cfreader.c:2635]: You may want to check `CFR_ACC_INTERFACE` and `CFR_ACC_MODULE`, as they are not allowed to have this attribute.

[the remark this answers is not in the document]
Comment (theresa-m) [reply runtime/bcutil/cfreader.c]: I have added this change

File: runtime/nls/cfre/cfrerr.nls [hunk 3]
```diff
@@ -1750,3 +1750,11 @@ J9NLS_CFR_ERR_NULLRESTRICTED_ATTRIBUTES_LENGTH_IS_ZERO.explanation=Please consul
 J9NLS_CFR_ERR_NULLRESTRICTED_ATTRIBUTES_LENGTH_IS_ZERO.system_action=The JVM will throw a verification or classloading related exception such as java.lang.ClassFormatError.
 J9NLS_CFR_ERR_NULLRESTRICTED_ATTRIBUTES_LENGTH_IS_ZERO.user_response=Contact the provider of the classfile for a corrected version.
 # END NON-TRANSLATABLE
+
+# ImplicitCreation is not translatable
+J9NLS_CFR_ERR_IMPLICITCREATION_ILLEGAL_CLASS_MODIFIERS=The ImplicitCreation attribute is only allowed in a non-abstract value class
```
Comment (hangshao0) [inline runtime/nls/cfre/cfrerr.nls:1755]: I see the message name is still `_ILLEGAL_CLASS_MODIFIERS`. As the message changed, I believe you need a new name for the message.

Comment (theresa-m) [reply runtime/nls/cfre/cfrerr.nls:1755]: To me it describes the gist of the issue, what would make it more clear?

Comment (hangshao0) [reply runtime/nls/cfre/cfrerr.nls:1755]: OK, I am good with it.

**Comment by hangshao0** [comment]
Jenkins test sanity,extended plinuxval jdknext

**Pull request merged**